src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -19,92 +19,135 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Configuration;
 using System.Diagnostics;
-using System.Web;
 using System.IO;
 using System.Text;
+using System.Web;
 
 public class PullRequestHandler : IHttpHandler
 {
-    private const string _gitExecutablePath = @"C:\Program Files\Git\cmd\git.exe";
-    private static string _gitRepoPath;
-    private const int _gitExecutionTimeoutMilliseconds = 100000;
-
-    public void ProcessRequest(HttpContext context)
-    {
-        // Check for prerequisites needed before this script can execute
+    private string _gitExecutablePath;
+    private string _gitRepositoriesBasePath;
+    private TimeSpan _gitExecutionTimeout;
 
-        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+    private bool _configurationRead = false;
 
-        StringBuilder result = new StringBuilder();
-        bool success = CheckPrerequisites(result);
+    private const string _gitExecutablePathConfigurationKeyString = "GitExecutablePath";
+    private const string _gitRepositoriesBasePathConfigurationKeyString = "GitRepositoriesBasePath";
+    private const string _gitExecutionTimeoutConfigurationKeyString = "GitExecutionTimeout";
 
-        if (!success)
-        {
-            // CheckPrerequisites failed, this is a server issue
-            context.Response.StatusCode = 500;
-            context.Response.Write(HttpUtility.HtmlEncode(result));
-            return;
-        }
+    private const string _gitRepoPathTemplate = @"{0}\wcf{1}";
 
-        success = CleanupMergedBranches(result);
+    public void ProcessRequest(HttpContext context)
+    {
+        StringBuilder result = new StringBuilder();
 
+        // Read configuration
+        bool success = ReadConfiguration(result);
         if (!success)
         {
-            // CleanupMergedBranches failed, this is a server issue
+            // ReadConfiguration failed, this is a server issue 
             context.Response.StatusCode = 500;
```

Comment:
I would have preferred to use 400.xxx, but unfortunately, this sub status code doesn't seem to be surfaced. 

So what we can do is to the StatusDescription to surface up a better message. Instead of just "bad request" we could reply with "bad request - PRService <blahblahblah>" if we need to. 

This is probably best done as a separate small discussion - changing the error codes and descriptions is trivial later on. #1059 created to track this work. I will leave the status code return values as-is for now and take #1059 as an improvement to the PRService.